src/scheduler/workers/unpublish/weekly/findNonRespondentsForList.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from "server/models/db/prisma-client";
+import { ListJsonData } from "server/models/types";
+import { schedulerLogger } from "scheduler/logger";
+import { List, Prisma } from "@prisma/client";
+import { findReminderToSend } from "./findReminderToSend";
+
+export async function findNonRespondentsForList(list: List) {
+  const logger = schedulerLogger.child({ listId: list.id, method: "findNonRespondentsForList" });
+
+  const jsonData = list.jsonData as ListJsonData;
+  const { keyDates } = jsonData.currentAnnualReview!;
+  const { unpublished } = keyDates;
+  logger.info(`unpublish date ${unpublished.UNPUBLISH}`);
+
+  const { reminderToFind, weeksSinceStartDate } = findReminderToSend(list);
+  const annualReviewDate = new Date(list.nextAnnualReviewStartDate!).toISOString();
+
+  if (weeksSinceStartDate >= 6) {
+    return [];
+  }
+
+  const editedSinceAnnualReviewDate: Prisma.EventWhereInput = {
+    type: "EDITED",
+    time: {
+      gte: annualReviewDate,
+    },
+  };
+
+  const reminderHasBeenSent: Prisma.EventWhereInput = {
+    type: "REMINDER",
```

Comment:
Indeed - but "in real life", if you've been sent a reminder on the 3rd week, and it's currently day 2 of week 3. and someone has screwed with the scheduler.. do you really want them to be sent another email? 

I think the way to test is to just gather up the events that happened at week X, do a count (N emails were sent on this date) and then purge them, The code is working as expected. 

I was conscious in making it at least unit testable, so that I am actually able to test by mocking system time. 

I don't think labelling a reminder that is sent 5 times is useful.

Reply:
Real life functioning is fine.  I'm thinking more for testing via the UI.  The purge approach, are you thinking that's a manual thing devs do when testing?  Ros will need a dev to test if so.